sdk/storage/azure_storage_blob/src/models/container_properties.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::{
+    headers::{
+        FromHeaders, HeaderName, Headers, ETAG, HAS_IMMUTABILITY_POLICY, HAS_LEGAL_HOLD,
+        LEASE_STATE, LEASE_STATUS, VERSION,
+    },
+    Error, Etag, LeaseStatus,
+};
+use typespec_client_core::fmt::SafeDebug;
+
+use crate::models::LeaseState;
+
+pub const LAST_MODIFIED: HeaderName = HeaderName::from_static("last-modified");
+pub const IMMUTABLE_STORAGE_WITH_VERSIONING_ENABLED: HeaderName =
+    HeaderName::from_static("x-ms-immutable-storage-with-versioning-enabled");
+
+/// Properties of an Azure Storage container.
+///
+#[derive(Clone, Default, SafeDebug)]
+pub struct ContainerProperties {
+    pub last_modified: Option<String>,
+    pub lease_state: Option<LeaseState>,
+    pub lease_status: Option<LeaseStatus>,
+    pub has_immutability_policy: Option<bool>,
+    pub has_legal_hold: Option<bool>,
+    pub immutable_storage_with_versioning_enabled: Option<String>,
+    pub etag: Option<Etag>,
+    pub version: Option<String>,
+}
+
+impl FromHeaders for ContainerProperties {
+    type Error = Error;
+    fn header_names() -> &'static [&'static str] {
+        &[
+            "etag",
+            "last-modified",
+            "x-ms-lease-state",
+            "x-ms-lease-status",
+            "x-ms-immutable-storage-with-versioning-enabled",
+            "x-ms-has-immutability-policy",
+            "x-ms-version",
```

Comment:
Version is not a container property, we should remove this and all its parsing logic.